subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -451,17 +582,25 @@ static void avdtp_process_configuration_cmd(struct bt_avdtp *session, struct net
 			err = -ENOTSUP;
 			avdtp_err_code = BT_AVDTP_BAD_STATE;
 		} else if (buf->len >= 1U) {
-			uint8_t int_seid;
-
+			uint8_t int_seid = 0;
 			/* INT Stream Endpoint ID */
-			int_seid = net_buf_pull_u8(buf) >> 2;
-
 			if (!reconfig) {
-				err = session->ops->set_configuration_ind(session, sep, int_seid,
-									  buf, &avdtp_err_code);
+				/* int seid not in reconfig cmd*/
+				int_seid = net_buf_pull_u8(buf) >> 2;
+			}
+			net_buf_simple_save(&buf->b, &state);
+			err = bt_avdtp_check_service_category(buf, &service_category, reconfig);
```

Comment:
suggest to use another variable (like: `err_code`) because `err` is `int`, but `bt_avdtp_check_service_category` return `uint8_t`.